js/tabs.js

```diff
@@ -322,6 +309,48 @@ Tabs.prototype.openFiles = function() {
       this.openFileEntry.bind(this));
 };
 
+/**
+ * @param {function()} callback
+ * Invoke the save dialog for all tabs with unsaved progress. Does not close any tabs.
+ */
+Tabs.prototype.promptAllUnsaved = function(callback) {
+  this.promptAllUnsavedFromIndex_(0, callback);
+}
+
+Tabs.prototype.promptAllUnsavedFromIndex_ = function(i, callback) {
+  if (i >= this.tabs_.length) {
+    callback();
+    return;
+  }
+  var tab = this.tabs_[i];
+  if (tab.isSaved()) {
+    this.promptAllUnsavedFromIndex_(i + 1, callback);
+  } else {
+    this.showTab(this.tabs_[i].getId());
+    this.promptSave_(tab, function(answer) {
+      if (answer === 'yes') {
```

Comment:
I don't know how this works, but can answer ever be not 'yes' because of localization?

Reply:
hmm good question. seems like it's all good, with the literal id 'yes' being set in DialogController.addButton, then this id being passed as the 'answer' to callback in DialobController.onClick_